include/cuco/static_set_ref.cuh

```diff
@@ -198,6 +198,22 @@ class static_set_ref
   template <typename... NewOperators>
   [[nodiscard]] __host__ __device__ auto with(NewOperators... ops) && noexcept;
```

Comment:
There's a [\deprecated](https://www.doxygen.nl/manual/commands.html#cmddeprecated) keyword in Doxygen which we can add to the old function to inform users that it will be deleted soon.